sources/mongodb/helpers.py

```diff
@@ -198,13 +205,89 @@ def load_documents(self, limit: Optional[int] = None) -> Iterator[TDataItem]:
             yield document
 
 
+class CollectionArrowLoader(CollectionLoader):
+    """
+    Mongo DB collection loader, which uses
+    Apache Arrow for data processing.
+    """
+
+    def load_documents(self, limit: Optional[int] = None) -> Iterator[Any]:
+        """
+        Load documents from the collection in Apache Arrow format.
+
+        Args:
+            limit (Optional[int]): The number of documents to load.
+
+        Yields:
+            Iterator[Any]: An iterator of the loaded documents.
+        """
+        context = PyMongoArrowContext.from_schema(
+            None, codec_options=self.collection.codec_options
+        )
+
+        cursor = self.collection.find_raw_batches(
+            self._filter_op, batch_size=self.chunk_size
+        )
+        if self._sort_op:
+            cursor = cursor.sort(self._sort_op)  # type: ignore
+
+        cursor = self._limit(cursor, limit)  # type: ignore
+
+        for batch in cursor:
+            process_bson_stream(batch, context)
+
+            table = context.finish()
+            new_schema = table.schema
+            for i, field in enumerate(table.schema):
+                if pymongoarrow.types._is_objectid(field.type):
+                    new_schema = new_schema.set(
+                        i, pyarrow.field(field.name, pyarrow.binary())
+                    )
+
+            table = table.cast(new_schema)
+
+            yield table
+
+
+class CollectionArrowLoaderParallel(CollectionLoaderParallel):
+    """
+    Mongo DB collection parallel loader, which uses
+    Apache Arrow for data processing.
+    """
+
+    def _get_cursor(self) -> TCursor:
+        cursor = self.collection.find_raw_batches(
+            filter=self._filter_op, batch_size=self.chunk_size
+        )
+        if self._sort_op:
+            cursor = cursor.sort(self._sort_op)  # type: ignore
+
+        return cursor
+
+    @dlt.defer
+    def _run_batch(self, cursor: TCursor, batch: Dict[str, int]) -> TDataItem:
+        cursor = cursor.clone()
+
+        context = PyMongoArrowContext.from_schema(
+            None, codec_options=self.collection.codec_options
+        )
+
+        for chunk in cursor.skip(batch["skip"]).limit(batch["limit"]):
+            process_bson_stream(chunk, context)
+
+            table = context.finish().to_pylist()
+
+            yield map_nested_in_place(convert_mongo_objs, table)
```

Comment:
why this is yielding objects while the regular arrow yields tables? we need arrow version of `convert_mongo_objs` (which you started writing above for the object id) and you should use it in both places

Reply:
Yeah, it's because I pushed changes to show you an approach. It actually wasn't finished.